include/seqan3/io/detail/take_until_view.hpp

```diff
@@ -0,0 +1,684 @@
+
+// ============================================================================
+//  detail::take_until_and_consume (adaptor instance definition)
+// ============================================================================
+
+/*!\brief A view adaptor that returns elements from the underlying range until the functor evaluates to true
+ *        (or the end of the underlying range is reached and consumes the end in single-pass ranges).
+ * \throws seqan3::unexpected_end_of_input If the underlying range contains no element that satisfies the functor.
+ * \ingroup views
+ *
+ * \copydetails seqan3::detail::take_until
+ * \hideinitializer
+ */
+inline auto constexpr take_until_and_consume = take_until_fn<false, true>{};
+
+// ============================================================================
+//  detail::take_until_or_throw_and_consume (adaptor instance definition)
+// ============================================================================
+
+/*!\brief A view adaptor that returns elements from the underlying range until the functor evaluates to true
+ *        (**throws** if the end of the underlying range is reached and consumes the end in single-pass ranges).
+ * \throws seqan3::unexpected_end_of_input If the underlying range contains no element that satisfies the functor.
+ * \ingroup views
+ *
+ * \copydetails seqan3::detail::take_until_and_consume
+ * \hideinitializer
+ */
+inline auto constexpr take_until_or_throw_and_consume = take_until_fn<true, true>{};
+
+//!\}
+
+} // namespace seqan3::detail
+
+namespace seqan3::views
+{
+
+/*!\brief A view adaptor that returns elements from the underlying range until the functor evaluates to
+ *        true (or the end of the underlying range is reached).
+ * \ingroup io
+ * \deprecated Use std::views::take_while with std::not_fn instead.
+ */
+SEQAN3_DEPRECATED_310 inline auto constexpr take_until = detail::take_until;
+
+/*!\brief A view adaptor that returns elements from the underlying range until the functor evaluates to true
+ *        (**throws** if the end of the underlying range is reached).
+ * \throws seqan3::unexpected_end_of_input If the underlying range contains no element that satisfies the functor.
+ * \ingroup io
+ * \deprecated Use std::views::take_while with std::not_fn instead.
+ */
+SEQAN3_DEPRECATED_310 inline auto constexpr take_until_or_throw = detail::take_until_or_throw;
+
+/*!\brief A view adaptor that returns elements from the underlying range until the functor evaluates to true
+ *        (or the end of the underlying range is reached and consumes the end in single-pass ranges).
+ * \throws seqan3::unexpected_end_of_input If the underlying range contains no element that satisfies the functor.
+ * \ingroup views
+ *
+ * \deprecated A previous call `view | take_until_and_consume(is_space)` could be convert to
+ *             `view | std::take_while(std::not_fn(is_space)) | std::drop_while(is_space)`
```

Comment:
```suggestion
 * \deprecated Use `std::take_while(std::not_fn(functor)) | std::drop_while(functor)` instead of
 *             `seqan3::views::take_until_and_consume(functor)`.
```

For me, this would be more helpful, because it reuses the `functor` which is mentioned in the brief, and technically you don't need the `view |`.

Could you adjust the other ones?
`take_until` should also get the "long" deprecation documentation

Reply:
Is this correct though?

I thought this does the stuff in sequence. First take everything until the first space and after that skip all spaces.

Reply:
Hah, right. Is there even an easy replacement?

Reply:
No there isn't if it isn't an input iterator.
I would just write that we don't have a replacement for this.

Reply:
Thanks for catching this! I remember Marcel and I already talked about this not being an appropriate replacement.
I now replaced the deprecation warning with "No alternative" as it was done in "as_const.hpp".